runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java

```diff
@@ -299,6 +299,66 @@ public Response submitEvent(@PathParam("machineId") String machineId,
         return Response.status(Response.Status.ACCEPTED).build();
     }
 
+    /**
+     * Update EventData of the specified Event name under the specified State machine
+     *
+     * @param machineId the state machine identifier
+     * @return Response with execution status code
+     * @throws Exception
+     */
+    @POST
+    @Path("/{machineId}/context/eventupdate")
+    @Timed
+    public Response updateEvent(@PathParam("machineId") String machineId,
+                                EventData eventData
+    ) throws Exception {
+        try {
+            LoggingUtils.registerStateMachineIdForLogging(machineId);
+            if (eventData.getData() == null || eventData.getName() == null || eventData.getEventSource() == null) {
+                return Response.status(Response.Status.BAD_REQUEST.getStatusCode()).entity(
+                        "Event Data|Name|Source cannot be null.").build();
+            }
+            if (eventsDAO.findBySMIdAndName(machineId, eventData.getName()).getStatus() != Event.EventStatus.triggered) {
+                return Response.status(Response.Status.FORBIDDEN.getStatusCode()).entity(
+                        "Input event is not in triggered state.").build();
+            }
+            boolean canUpdateEventData = false;
+            /* List holds objects retrieved from States matching input machineId and eventName as [taskId, machineId, status] */
+            List<Object[]> states = statesDAO.findStatesByDependentEvent(machineId, eventData.getName());
+
+            for (Object[] state : states) {
+                if (state[2] == Status.running) {
+                    canUpdateEventData = false;
+                    break;
+                } else if (state[2] == Status.initialized || state[2] == Status.errored || state[2] == Status.sidelined) {
+                    canUpdateEventData = true;
+                }
+            }
+            if (canUpdateEventData) {
+                this.workFlowExecutionController.persistEvent(machineId, eventData);
+
+                /* Audit entry in AuditRecord. Though audit record is in alignment with stateId, adding here
```

Comment:
Put these steps in a separate function and add these comments over that function.

Reply:
Done.